lib/token-downloader.js

```diff
@@ -0,0 +1,72 @@
+"use strict";
+
+
+/**
+ * @typedef {Object} tokenDownloader
+ * @function download
+ * @function downloadToFile
+ */
+
+
+/**
+ * Creates tokenDownloader class.
+ *
+ * @param {bluebird} bluebird
+ * @param {fs} fs
+ * @param {request} request
+ * @param {request} requestAsync
```

Comment:
These two lines are not injecting the same thing.

Technically, you don't need request if you have requestAsync.